Week-17/Router/IncomeExpenseRouter.js

```diff
@@ -0,0 +1,270 @@
+let express = require('express');
+let incomeExpenseRouter = express.Router();
+let User = require('../Model/UserSchema');
+
+incomeExpenseRouter.get('/', async (req, res) => {
+    if (req.query.year || req.query.month) {
+        let savings = [];
+        let incomes = [];
+        let expenses = [];
+        if (req.query.year) {
+            incomes = await User.aggregate([
+                { $match: { _id: req.rootUser._id } }, 
+                { $unwind: "$incomes" }, 
+                { $match: { "incomes.year": req.query.year } }, 
+                { $group: { _id: "", totalIncomes: { $sum: "$incomes.amount" } } }, 
+                { $project: { _id: 0, totalIncomes: "$totalIncomes" } }
+            ]);
+            expenses = await User.aggregate([
+                { $match: { _id: req.rootUser._id } }, 
+                { $unwind: "$expenses" }, 
+                { $match: { "expenses.year": req.query.year } }, 
+                { $group: { _id: "", totalExpenses: { $sum: "$expenses.amount" } } }, 
+                { $project: { _id: 0, totalExpenses: "$totalExpenses" } }
```

Comment:
Can we have a common method to get this array? Both arguments look very similar, can we think of a way to write them in such a way that we have to write it only once?

Reply:
Sure can create a common function for it.